osu.Game/Beatmaps/Drawables/Cards/BeatmapCard.cs

```diff
@@ -24,6 +25,8 @@ public abstract partial class BeatmapCard : OsuClickableContainer, IHasContextMe
 
         protected const float WIDTH = 345;
 
+        public CollapsibleButtonContainer ButtonContainer { get; set; } = null!;
```

Comment:
I _really really_ dislike this. This is how multiplayer ended up in the place it currently is, where [random things](https://github.com/ppy/osu/blob/73adc4dfd7bcc4546c3e98f045d0346581f82f16/osu.Game/Screens/OnlinePlay/Match/RoomSubScreen.cs#L52) are exposed for the purpose of inheriting screens setting them that eventually goes completely [out of control](https://github.com/smoogipoo/osu/blob/be33addae16f589dda941d27d2e49a25ec61d0bd/osu.Game/Screens/OnlinePlay/Match/RoomSubScreen.cs#L48-L63).

Let's not make the same mistake again.

Reply:
I agree it's not 100% optimal, but also can't read your mind 😅.

I also tried an `Action RequestDownload` kinda flow but it felt way too verbose for what this is. Would that work better?

What's your preference?

Reply:
@smoogipoo prod on this one, need some direction before attempting something else that doesn't work for you.

Reply:
Sorry, I thought this was in response to the comment below.

Here's two suggestions:
1. Make `ContextMenuItems` virtual and duplicate the implementation.
2. Add abstract `DownloadButton? GetDownloadButton()` and implement in every place.